core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp

```diff
@@ -397,6 +397,27 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
                      FunctorTeamShmemSize<FunctorType>::value(
                          arg_functor, arg_policy.team_size())) {
     m_instance = arg_policy.space().impl_internal_space_instance();
+
+    if ((arg_policy.scratch_size(0) +
+         FunctorTeamShmemSize<FunctorType>::value(arg_functor,
+                                                  arg_policy.team_size())) >
+        static_cast<size_t>(TeamPolicy<Kokkos::OpenMP>::scratch_size_max(0))) {
```

Comment:
`scratch_size` returns `size_t` but `scratch_size_max` returns `int`. We might want to change that to be consistent.